app/helpers/availabilities_helper.rb

```diff
@@ -2,19 +2,62 @@
 
 module AvailabilitiesHelper
   def generate_cell(day, time_range)
-    generate(@availabilities[[day, time_range.id]],
-             calc_colspan(time_range.start_time, time_range.end_time))
+    generate(@availabilities[[day, time_range.id]])
   end
 
-  def generate(availability, col_span)
+  def generate(availability)
     id = availability.id
-    td_attributes = { align: 'center', class: 'checkbox_cell',
-                      onclick: "toggle(#{id})", id: "cell_#{id}",
-                      colspan: col_span }
+    check_box_tag("availability_ids[#{id}]", id, availability.status,
+                  style: 'visibility: hidden;')
+  end
 
-    content_tag :td, td_attributes do
-      check_box_tag "availability_ids[#{id}]", id, availability.status,
-                    style: 'visibility: hidden; display: none:'
+  def stylise_user(user)
+    if user[:mc]
+      content_tag(:strong, user[:username])
+    else
+      user[:username]
     end
   end
+
+  def generate_select(ts)
+    day_index = Availability.days[ts.day]
+    time_range_id = ts.time_range_id
+    current_users = change(day_index, time_range_id,
+                           (ts.mc_only ? @users.select(&:mc) : @users))
+    select_tag("#{day_index % 7}#{time_range_id}",
+               options_from_collection_for_select(
+                 current_users,
+                 'id', 'username', selected: ts.default_user_id
+               ), class: 'availability-select')
+  end
+
+  def change(day_id, timerange_id, users)
+    users.map do |u|
+      OpenStruct.new(id: u.id, username: u.username +
+        (@availabilities[[day_id, timerange_id]].include?(u.id) ? ' ✓' : ''))
+    end
+  end
+
+  def calc_offset(tr)
```

Comment:
Not clear what the constants `80` `40` etc are. Probably better to abstract them out into a constant

Reply:
Done